### PR TITLE
Swe/filter transactions02

### PR DESCRIPTION
**Pull request by sprucely**
Allow transaction filtering by merchant id and tracking number.

### CODE REVIEW

**Comment by sprucely** [comment]
The transaction number search is now on a separate page and including related records when a processor id is matched.

I also changed calls I had added in previous commits to window.location.replace with window.location.assign because it turned out to cause unintuitive results with the back button.

**Comment by jSylvestre** [comment]
@sprucely I meant something like this:
image


Which you can do by adding this code to the layout page:

`                    @if (!string.IsNullOrEmpty(teamSlug))
                    {
                        <li class="nav-item">
                            <form class="topsearch flex-grow-1 mr-sm-4" asp-controller="Transactions" asp-action="Search" asp-route-team="@teamSlug" method="get">
                                <input class="form-control" name="trackingNum" type="search" placeholder="Search Transactions" value="@query" aria-label="Search">
                            </form>
                        </li>
                    }`

**Comment by sprucely** [comment]
Oh, I see. No prob...

**Comment by sprucely** [comment]
Okay, how's this look?

**Comment by jSylvestre** [comment]
@sprucely Just have the placeholder text be Search Transactions. It is getting chopped:
Also, you may want to put it to the right of the jobs menu.

image

**Comment by jSylvestre** [comment]
You may have also toasted this. I can't remember if I saw it before:

image

File: Sloth.Web/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -80,11 +97,32 @@
                         </li>
                     }
                 </ul>
-                <partial name="_LoginPartial" />
```
Comment (jSylvestre) [inline Sloth.Web/Views/Shared/_Layout.cshtml:83]: @sprucely Need this back :)

**Comment by sprucely** [comment]
@jSylvestre @srkirkland Please take a look. It's now much more consistent with how filtering is done in Payments; scrapped most of the javascript.

**Comment by srkirkland** [comment]
To add a little, session is problematic in a cloud environment even with edge affinity set. There are distributed ways to do it but generally it’s worth avoiding if possible.

**Comment by sprucely** [comment]
Sure. And just in case you weren't aware, Payments is storing filters in a session.
________________________________

**Comment by srkirkland** [comment]
I didn’t know that— that’s unfortunate. I’ll make an issue to look into that later.

**Comment by sprucely** [comment]
Here's the latest. No more sessions, and I eliminated several files, merging search functionality into a conditional branch in Index action and view.

**Comment by sprucely** [comment]
@srkirkland Should I address the lgtm alert before merging? It's not valid.

**Comment by srkirkland** [comment]
> @srkirkland Should I address the lgtm alert before merging? It's not valid.

Yea, might as well

**Pull request merged**